app/main.py

```diff
@@ -1489,35 +1585,43 @@ def update_alert_frame_main(alert_frame_update_new_event, alert_frame_update_int
 @app.callback(
     Output("dashboard_table", "children"),
     Input("interval-component-dashboard-screen", "n_intervals"),
+    [State("user_headers", "data"), State("user_credentials", "data")],
 )
-def update_dashboard_table(n_intervals):
+def update_dashboard_table(n_intervals, user_headers, user_credentials):
     """
     This builds and refreshes the dashboard table to monitor devices status every minute
     Had to fetch devices data again for freshness purposes
     """
-    response = requests.get("https://api.pyronear.org/devices/", headers=api_client.headers)
+    if user_headers is None:
+        raise PreventUpdate
+
+    response = requests.get("https://api.pyronear.org/devices/", headers=user_headers)
     # Check token expiration
     if response.status_code == 401:
-        api_client.refresh_token(cfg.API_LOGIN, cfg.API_PWD)
+        api_client.refresh_token(user_credentials["username"], user_credentials["password"])
         response = requests.get("https://api.pyronear.org/devices/", headers=api_client.headers)
 
     # We filters devices_data to only display devices belonging to the sdis and then make the comparison between
     # last_ping and datetime.now()
     all_devices = pd.DataFrame(response.json())
 
-    sdis_devices = (
-        all_devices[all_devices["id"].isin(range(2, 18))]
-        .sort_values(by="login")[["yaw", "lat", "lon", "login", "last_ping"]]
-        .copy()
-    )
+    if not all_devices.empty:
+        sdis_devices = all_devices.sort_values(by="login")[["yaw", "lat", "lon", "login", "last_ping"]].copy()
 
-    sdis_devices["last_ping_hours_dif"] = sdis_devices["last_ping"].apply(
-        lambda x: (pd.to_datetime(x) - pd.to_datetime(datetime.utcnow().isoformat())).total_seconds() // 3600
-    )
+        # For some Reolink devices, it seems that the last ping is missing
+        # We add this line to prevent these cases from breaking the logic below
+        sdis_devices = sdis_devices.dropna(subset=["last_ping"]).copy()
```

Comment:
I checked and it seems that last_ping can't be `null` as there is a default value taking the creation timestamp

Reply:
I do find a `None` last ping for two devices (IDs 31 and 32), which breaks the `datetime` conversion. The `software_hash` is also missing for these two. For now, the `drop_na` is still required for the callback to run 😢

Reply:
Those are tests made by @fe51 that seem kinda buggy to me 😅 

Ok to let this safety then !